src/bot/api.rs

```diff
@@ -1707,19 +1707,921 @@ impl Bot {
 impl Requester for Bot {
     type Err = crate::errors::RequestError;
 
+    type GetUpdates = JsonRequest<payloads::GetUpdates>;
+
+    fn get_updates(&self) -> Self::GetUpdates where {
+        Self::GetUpdates::new(self.clone(), payloads::GetUpdates::new())
+    }
+
+    type SetWebhook = JsonRequest<payloads::SetWebhook>;
+
+    fn set_webhook<U, A>(&self, url: U, allowed_updates: A) -> Self::SetWebhook
+    where
+        U: Into<String>,
+        A: IntoIterator<Item = crate::types::AllowedUpdate>,
+    {
+        Self::SetWebhook::new(self.clone(), payloads::SetWebhook::new(url, allowed_updates))
+    }
```

Comment:
Don't you think gathering associated types together (and the same for functions) would make the code cuter?

Reply:
I don't think so b/c the types and the methods are highly related to each other.